### PR TITLE
Fixed performance when there is a lot of item in the m2m relation and a visual glitch.

### PR DESCRIPTION
**Pull request by johnraz**
Formerly the jQuery code was looping through all the potential `<input>` elements
that could be added to the m2m relationship in order to find the parent
`<ul>` elements. It is now done this other way around, we only loop
through `<ul>` elements that have `<input>` child with the `sortedm2m`
class.

(With this change a page that would take 12sec to load before is now loading instantly.)

We are also hidding the `<ul>` until it gets fully prepared, to avoid to have
a bad flicker when initially loading the page.

### CODE REVIEW

**Comment by johnraz** [comment]
ping @gregmuellegger :-)

File: sortedm2m/templates/sortedm2m/sorted_checkbox_select_multiple_widget.html [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+
 <div class="sortedm2m-container">
 
     <p class="selector-filter">
         <img src="{% static "sortedm2m/selector-search.gif" %}" alt="" title="{% trans "Type into this box to filter down the list." %}" />
         <input type="text" placeholder="{% trans "Filter" %}" />
     </p>
-
-    <ul>
+    <ul class="hide">
```
Comment (gregmuellegger) [inline sortedm2m/templates/sortedm2m/sorted_checkbox_select_multiple_widget.html]: Could we make the `hide` class be added by JS on load? The problem here I see is if you don't have JS enabled you will never see the checkboxes ... I'm not sure if I'm living in the 1990s in that regard, what do you think?

**Comment by gregmuellegger** [comment]
Hey, that's a super cool addition. Thanks for taking the time! I will merge it as soon as I know your opinion on the inline comment I made ;)

**Comment by johnraz** [comment]
I don't believe anybody will hit django's admin interface without js enable nowadays but well, it's always better to be safe so I applied what you asked for ;-)

**Comment by gregmuellegger** [comment]
Awesome! Merged and released in 1.0.1 https://pypi.python.org/pypi/django-sortedm2m/1.0.1

**Comment by johnraz** [comment]
Sweet! thanks for the release ;-)

**Pull request merged**